addons/payment_razorpay/views/payment_provider_views.xml

```diff
@@ -11,17 +11,60 @@
                        invisible="code != 'razorpay'">
                     <field name="razorpay_key_id"
                            string="Key Id"
-                           required="code == 'razorpay' and state != 'disabled'"/>
+                           groups="base.group_no_one"/>
                     <field name="razorpay_key_secret"
                            string="Key Secret"
-                           required="code == 'razorpay' and state != 'disabled'"
-                           password="True"/>
-                    <field name="razorpay_webhook_secret"
-                           string="Webhook Secret"
-                           required="code == 'razorpay' and state != 'disabled'"
+                           groups="base.group_no_one"
                            password="True"/>
+                    <label for="razorpay_webhook_secret"/>
+                    <div class="o_row">
+                        <field name="razorpay_webhook_secret"
+                            string="Webhook Secret"
+                            groups="base.group_no_one"
+                            password="True"/>
+                        <button string="Create Webhook"
+                                name="action_razorpay_create_and_update_webhook"
+                                type="object"
+                                help="If you change the your base url, you will need to update the webhook. else payment state will not be updated in odoo."
+                                class="btn-primary"
+                                groups="base.group_no_one"
+                                invisible="not razorpay_access_token or razorpay_webhook_id"/>
+                        <button string="Update Webhook"
+                                name="action_razorpay_create_and_update_webhook"
+                                type="object"
+                                help="If you change the your base url, you will need to update the webhook. else payment state will not be updated in odoo."
+                                class="btn-primary"
+                                groups="base.group_no_one"
+                                invisible="not razorpay_access_token or not razorpay_webhook_id"/>
+                    </div>
                 </group>
             </group>
+            <group name="provider_credentials" position="after">
+                <field name="razorpay_public_token" invisible="1"/>
+                <field name="razorpay_refresh_token" invisible="1"/>
+                <field name="razorpay_account_id" invisible="1"/>
+                <field name="razorpay_access_token" invisible="1"/>
+                <field name="razorpay_access_token_expiration" invisible="1"/>
+                <field name="razorpay_key_id" invisible="1"/>
+                <field name="razorpay_webhook_secret" invisible="1"/>
+                <field name="razorpay_webhook_id" invisible="1"/>
```

Comment:
Instead of using invisible fields, we can use `fieldDependencies`. you can decide.

https://github.com/odoo/odoo/blob/0bc69d757f7c3429c6112aa5aaa1ceb999428189/addons/web/static/src/views/kanban/kanban_arch_parser.js#L136